DOCSP-48672 Document API Access (#982)
mongodb/docs-atlas-cli: source/atlas-cli-admin-api.txt

.. _atlas-cli-admin-api:

===============================================================

> Run {+atlas-cli+} Commands with {+atlas-admin-api+}

Reviewer: [suggestion] could we use a "the" before Atlas Admin API?

```suggestion
Run {+atlas-cli+} Commands with {+atlas-admin-api+}
```

===============================================================

.. meta::
   :description: Learn to execute Atlas CLI commands using the Atlas Administration API.

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

This tutorial shows you how to run {+atlas-cli+} commands with the
{+atlas-admin-api+}. You can use every {+atlas-admin-api+} resource
and endpoint from the {+atlas-cli+} to create and manage |service|

> deployments. {+atlas-cli+} support for {+atlas-admin-api+} provides the

Reviewer: ```suggestion
deployments. {+atlas-cli+} support for the {+atlas-admin-api+} provides the
```

options with the command. You can specify the options directly 
with the command or using a |json| file. The command also supports the 
following options: 

.. list-table:: 
   :widths: 20 10 70 
   :header-rows: 1

   * - Option 
     - Necessity 
     - Description

   * - ``--file`` 
     - Conditional 
     - |json| file that contains the |api| path, query, and request body
       parameters for the operation. This is required only if there are
       required path, query, or request body parameters for the operation. 

   * - ``--version`` 
     - Optional 
     - |api| resource version to use. We recommend using it to pin your
       scripts to specific |api| versions. If omitted, the command
       defaults to the latest version (or your profile's configured
       version). However, we recommend explicitly setting the version to
       ensure your scripts remain stable. This protects your scripts
       from breaking when new |api| versions are released with
       potentially incompatible changes.

.. _atlas-cli-admin-api-examples:

Examples 
--------

The following {+atlas-cli+} command with the {+atlas-admin-api+}

> demonstrates howto retrieve a compressed (``.gz``) log file that

Reviewer: ```suggestion
demonstrates how to retrieve a compressed (``.gz``) log file that
```

contains a range of log messages for the specified host for the
specified project:  

.. code-block:: shell 

   atlas api monitoringAndLogs getHostLogs --groupId 5e2211c17a3e5a48f5497de3 --hostName mycluster-shard-00-02.7hgjn.mongodb.net --logName mongodb --output gzip --version 2025-03-12

The following {+atlas-cli+} command with the {+atlas-admin-api+}
demonstrates how to create a cluster by using the ``--file`` option.

.. code-block:: shell 

   atlas api clusters createCluster --groupId 5e2211c17a3e5a48f5497de3 --file cluster-config.json --version 2025-03-12

To learn more about creating a configuration file for a cluster, see
:ref:`atlas-cli-cluster-config-file`. 

The following {+atlas-cli+} command with the {+atlas-admin-api+}

> demonstrates how to simulate simulate regional cloud provider outages.

Reviewer: ```suggestion
demonstrates how to simulate regional cloud provider outages.
```

This simulation lets you test your application's failover behavior and
disaster recovery procedures in a controlled environment separate from
production. The command uses a file named ``outage_simulation.json``
with the following settings: 

.. code-block:: 

   {
      "outageFilters": [
         {
            "cloudProvider": "AWS",
            "regionName": "US_EAST_1",
            "type": "REGION"
         }
      ]
   }

.. io-code-block:: 
   :copyable: true 

   .. input:: 
      :language: shell 

      atlas api clusterOutageSimulation startOutageSimulation --groupId 5e2211c17a3e5a48f5497de3 --clusterName myCluster --file outage_simulation.json --version 2025-03-12

   .. output:: 

>       :language: she;; 

Reviewer: ```suggestion
      :language: shell 
```

- A unified, predictable command structure for automation
- Ability to pin a desired API version, ensuring your scripts remain
  reliable, even if you update the CLI.

Syntax 
------

To use {+atlas-cli+} with the {+atlas-admin-api+}, run the command in
the following format: 

.. code-block:: shell 

   atlas api <tag> <operationId> [options]|--file <fileName>.json --version <api-resource-version>

Arguments 
~~~~~~~~~

.. list-table:: 
   :widths: 20 10 70 
   :header-rows: 1

   * - Argument 
     - Necessity
     - Description

   * - ``<tag>``
     - Required
     - The name of the tag used in the {+atlas-admin-api+} documentation
       |url| for the |api| resource. The tag is hyphen-separated in the
       {+atlas-admin-api+} documentation |url|. However, you must
       convert it to camelcase in the {+atlas-cli+} command syntax. 

       .. code-block:: shell 

          https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/<Tag-Name>/
       
       For accessing the resource in the preceding |url|, replace
       ``<tag>`` with ``tagName`` in the command: 

       .. code-block:: shell 

>           atlas admin tagName <operationId> 

Reviewer 2: This should be `atlas api tagName <operationId>`.

For consistency, it would maybe be better to use `<>` around `tagName` too.

Like this: `atlas api <tagName> <operationId>`

Author: This is an example. Fixed.
